tests/rptest/tests/group_membership_test.py

```diff
@@ -119,6 +121,76 @@ def _check_no_duplicates():
                    err_msg="found persistent duplicates in groups listing")
 
 
+class GroupCoordinatorTransferUtils:
+    """
+    Utilities class for handling the topic-partition for the __consumer_offsets topic.
+    This class assumes that the topic has 1 partition a replication factor of 3
+    """
+    def __init__(self, redpanda: RedpandaService):
+        self.redpanda = redpanda
+        self.admin = Admin(self.redpanda)
+        self.logger = self.redpanda.logger
+
+    def get_group_partition(self, partition: int = 0):
+        partition = self.admin.get_partitions(namespace="kafka",
+                                              topic="__consumer_offsets",
+                                              partition=partition)
+        self.logger.debug(f"Group partition: {partition}")
+        return partition
+
+    def get_group_leader(self):
+        leader = self.get_group_partition()['leader_id']
+        self.logger.debug(f"Group leader: {leader}")
+        return leader
+
+    def transfer_leadership(self, new_leader: ClusterNode):
+        """
+        Request leadership transfer of the internal consumer group partition
+        and check that it completes successfully.
+        """
+        self.logger.debug(
+            f"Transferring leadership to {new_leader.account.hostname}")
+        success = self.admin.transfer_leadership_to(
+            namespace="kafka",
+            topic="__consumer_offsets",
+            partition=0,
+            target_id=self.redpanda.idx(new_leader))
+
+        self.logger.debug(
+            f"Leadership transfer to {new_leader}, success: {success}")
+        return success
+
+    def validate_leadership_transfer(self, new_leader: ClusterNode):
+        for _ in range(3):  # re-check a few times
```

Comment:
nit: Would this be more naturally formulated as a `wait_until`?

Reply:
I don't think so. It seems to me that this for loop is intended to retry fetching the group leader. Additionally, this method is used in the `wait_until` call on line 398 so it would be strange to me to have a nested `wait_until` for this.

Reply:
I don't think my opinion has changed, am I missing something?

```python
        def leader_is(): 
            leader = self.get_group_leader()
            return leader != -1 and self.redpanda.get_node(leader) == new_leader
     
        wait_until(leader_is, timeout_sec=4, backoff_sec=1)
```
Anyway, it's a nit since it's just moved code.